eng/scripts/Merge-Branch.ps1

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Use case: This script merges changes from a source branch into the current branch with the behavior:
+# 1. Overwrite paths matching $Theirs (this includes deletes)
+# 2. Ensure files in $Ours remain untouched
+# 3. For paths matching $Merge, merge changes from $SourceBranch allowing the user to resolve conflicts manually
+#
+# Adding paths to $Merge excludes them from the default keep or overwrite behaviour of $Ours and $Theirs.
+#
+# This script can be run locally from the root of the repo:
+# .\eng\scripts\Merge-Branch.ps1 -SourceBranch 'main' -Theirs '**' -Ours 'sdk/template' -Merge 'sdk/template/ci.yml', '**/README.md'
+#
+# This would merge main into the local branch, making the working folder look like main. It not overwrite sdk\template.
+# Changes in sdk\template\ci.yml and readme.md files would be merged, not excluded or overwritten.
+
+[CmdLetBinding()]
+param(
+    [string]$SourceBranch,
+    [string[]]$Theirs, # paths to always overwrite
+    [string[]]$Ours, # paths to never merge or overwrite
+    [string[]]$Merge # paths to merge or overwrite
+)
+
+$theirIncludes = @($Theirs | ForEach-Object { ":(top,glob)$_" })
```

Comment:
What are these magic include filters? Can you link to the git documentation covering them in a comment?

Reply:
Comments added to script in pr #31222